compiler/rustc_passes/src/check_attr.rs

```diff
@@ -582,17 +590,38 @@ impl CheckAttrVisitor<'tcx> {
         if &*item_name.as_str() == doc_alias {
             return err_fn(meta.span(), "is the same as the item's name");
         }
+        let span = meta.span();
+        if aliases.try_insert(doc_alias.to_owned(), span).is_err() {
+            self.tcx.struct_span_lint_hir(UNUSED_ATTRIBUTES, hir_id, span, |lint| {
+                lint.build("doc alias is duplicated")
+                    .span_label(*aliases.get(doc_alias).unwrap(), "first defined here")
```

Comment:
nit: you don't need the second lookup here, right? You can do `if let Err(entry) = ...` above and then `*entry.get()` here?

Reply:
Absolutely! Great catch!

Reply:
Ah actually no, I did read it well the first time: it returns the non-inserted value, which is useless in here.

Reply:
*entry.entry.get() should still work: the returned Err contains both the value passed in and the hashmap entry you would have overwritten with a regular insert.

Reply:
Damn, didn't pay attention to that. This API is even better than what I originally thought!